### PR TITLE
Add support for sanitization pipeline

### PR DESCRIPTION
**Pull request by Ndpnt**
Also cleanup facebook docs

### CODE REVIEW

File: src/sanitizer/sanitizers.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export function cleanFacebookUrls(document) {
```
[the remark this answers is not in the document]
Comment (Ndpnt) [reply src/sanitizer/sanitizers.js:1]: what about `src/sanitizer/sanitizers/facebook.js` or `src/sanitizer/service_providers/facebook.js`?

File: src/sanitizer/index.js [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 
+import * as sanitizers from './sanitizers.js';
+
 const { JSDOM } = jsdom;
 const turndownService = new TurndownService();
 
-export default async function sanitize(content, selector) {
+export default async function sanitize(content, selector, sanitizationPipeline) {
   let contentToSanitize = content;
+  const { document } = new JSDOM(contentToSanitize).window;
```
[the remark this answers is not in the document]
Comment (Ndpnt) [reply src/sanitizer/index.js:11]: Yep I agree and I think in 99% of the time we will a t least have a selector.

File: src/sanitizer/sanitizers.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (MattiSG) [reply src/sanitizer/sanitizers.js:1]: As discussed: `providers/facebook.sanitizers.js`

File: src/index.js [hunk 3]
```diff
@@ -7,14 +7,14 @@ consoleStamp(console);
 import scrape from './scraper/index.js';
 import { persistRaw, persistSanitized, pushChanges } from './history/index.js';
 import sanitize from './sanitizer/index.js';
-import getServiceProviders from './service_providers/index.js';
+import getServiceProviders, { getSanitizers } from './service_providers/index.js';
 import { DOCUMENTS_TYPES } from './documents_types.js';
 
-export async function updateServiceProviderDocument(serviceProviderId, serviceProviderName, documentType, documentUrl, documentContentSelector) {
+export async function updateServiceProviderDocument({ serviceProviderId, serviceProviderName, documentType, documentParams }) {
```
Comment (MattiSG) [inline src/index.js:13]: To me this is not the document _parameters_ but really the _document_ itself —or maybe the _document descriptors_?

```suggestion
export async function updateServiceProviderDocument({ serviceProviderId, serviceProviderName, documentType, document }) {
```

File: src/index.js [hunk 4]
```diff
@@ -39,14 +40,18 @@ export default async function updateTerms() {
...
   Object.keys(serviceProvidersManifests).forEach((serviceProviderId) => {
     const { documents, serviceProviderName } = serviceProvidersManifests[serviceProviderId];
 
     Object.keys(documents).forEach((documentType) => {
-      const { url, contentSelector } = documents[documentType];
-      documentUpdatePromises.push(updateServiceProviderDocument(serviceProviderId, serviceProviderName, documentType, url, contentSelector));
+      documentUpdatePromises.push(updateServiceProviderDocument({
+        serviceProviderId,
+        serviceProviderName,
+        documentType,
```
Comment (MattiSG) [inline src/index.js:52]: > Maybe move this into the `documentParams`?

File: src/sanitizer/index.js [hunk 5]
```diff
@@ -4,11 +4,18 @@ import jsdom from 'jsdom';
 const { JSDOM } = jsdom;
 const turndownService = new TurndownService();
 
-export default async function sanitize(content, selector) {
+export default async function sanitize(content, selector, sanitizationPipeline, additionalSanitizers) {
```
Comment (MattiSG) [inline src/sanitizer/index.js:7]:
```suggestion
export default async function sanitize(content, selector, sanitizationPipeline, providerSpecificSanitizers) {
```

File: src/service_providers/index.js [hunk 6]
```diff
@@ -6,18 +6,26 @@ dotenv.config();
...
-    const serviceProviderId = path.basename(filename, '.json');
-    if (serviceProviderId.indexOf('.') === 0) {
-      return;  // ignore invisible files such as .DS_Store
+  const filenames = fs.readdirSync(dirPath);
+
+  for (let filename of filenames) {
+    if (filename.indexOf('.sanitizers.js') !== -1) {
+      const serviceProviderId = path.basename(filename, '.sanitizers.js');
+      sanitizers[serviceProviderId] = await import(path.join(dirPath, filename));
+    } else if (filename.indexOf('.json') !== -1) {
```
Comment (MattiSG) [inline src/service_providers/index.js:20]: > We might end up with a service provider name collision here at some point, as we're not explicitly testing the extension.

**Pull request merged**